src/core/server/config/config_service.ts

```diff
@@ -148,7 +194,21 @@ export class ConfigService {
     return config.getFlattenedPaths().filter(path => isPathHandled(path, handledPaths));
   }
 
-  private validate(path: ConfigPath, config: Record<string, unknown>) {
+  private async logDeprecation() {
+    const rawConfig = await this.rawConfigProvider
+      .getConfig$()
+      .pipe(take(1))
+      .toPromise();
+    const deprecations = await this.deprecations.pipe(take(1)).toPromise();
+    const deprecationMessages: string[] = [];
+    const logger = (msg: string) => deprecationMessages.push(msg);
+    applyDeprecations(rawConfig, deprecations, logger);
+    deprecationMessages.forEach(msg => {
+      this.log.warn(msg);
+    });
```

Comment:
Should I use a more specific logger than the root `ConfigService` logger? Could use https://github.com/elastic/kibana/issues/39695